baseplate/integration/pyramid.py

```diff
@@ -86,6 +86,45 @@ class ServerSpanInitialized(BaseplateEvent):
     pass
 
 
+class BaseplateHeaderTrustHandler(object):
+
+    """Class used by BaseplateConfigurator to validate trace and edge context headers. Specifically
+    implements hmac validation for the trace headers.
+
+    param bool trust_trace_headers: Whether to trust trace headers by default
+    param bool trust_edge_context_header: Whether to trust the edge context header by default
+    param baseplate.secrets.SecretsStore secret_store: secret store which should contain a
+        versioned secret at the path trace/hmac if doing hmac validation of the trace headers.
+    """
+    def __init__(
+        self,
+        trust_trace_headers=False,
+        trust_edge_context_header=False,
+        secret_store=None
+    ):
+        self.trust_trace_headers = trust_trace_headers
+        self.trust_edge_context_header = trust_edge_context_header
+        self.secret_store = secret_store
```

Comment:
I can see your point. What I may propose then is that we would could move the BaseplateHeaderTrustHandler to a separate file, and maybe rename to HmacHeaderTrustHandler. That way it can be reused by gateway and other services that need it. We can then create the default version here which simply uses the `trust_trace_headers` flag.

Reply:
They can be in the same file or a separate one, that shouldn't be an issue for re-use.  I think we would want something like:

```
class HeaderTrustHandlerInterface
class StaticTrustHandler(HeaderTrustHandlerInterface)
class HmacSignatureTrustHandler(HeaderTrustHandlerInterface)
```

Reply:
Yeah, I meant two unrelated things there. Mainly that we should create the new HmacSignatureTrustHandler class here in baseplate and not in Gateway for reuse, but also moving to a separate spot for code organization purposes.

Reply:
I definitely agree with @pacejackson. I think a dependency injection type approach is best -- construct the Trust object that you want and pass it into the configurator.

Reply:
Yeah, this would be a nice clean approach and let's us unify the `trust_trace_headers` boolean into the same system. The existing flag can be maintained at the external API, given a deprecation warning, and would just add the equivalent "handler" to the list.